Dockerfile

```diff
@@ -1,8 +1,27 @@
 ARG ARCH="amd64"
 ARG OS="linux"
+FROM quay.io/prometheus/golang-builder:main as builder
```

Comment:
i wonder if we should pin to a specific major version in line with the go.mod?
The GitHub runner already has access to the ${golang-version} variable defined in `.github/env`.

Reply:
I didn't because I figured the lattest go version will (or should) always be able to build projects that specify an older version in `go.mod`.
Afaiu Prometheus only keeps the builders up to date for n-1 versions because they test with the previous version. Unless p-o does something similar, I don't see an advantage to pining a specific builder version.

Reply:
Yeah I don't have a concrete example which would require pinning the Go version but I'd prefer if we ensure that the same Go version is used everywhere. Can we have a way to inject the Go image tag and default to main if unspecified?

Reply:
How about this?
Added a new make target that updates the version as specified in `.github/env` (maybe use go.mod instead?) and integrate it into the generate target.

Reply:
Oh even better...those are much smaller.